spec/mem/tree_borrows/tree.md

```diff
@@ -76,11 +79,14 @@ impl Node {
     ) -> Result {
         let offset_start = offset_in_alloc.bytes();
         for offset in offset_start..offset_start + size.bytes() {
-            self.location_states.mutate_at(offset, |location_state|{
-                location_state.transition(access_kind, node_relation, self.protected.yes())
+            self.permissions.mutate_at(offset, |permission|{
+                permission.transition(access_kind, node_relation)
             })?;
         }
 
+        // Re-assert the invariant that our state machine corresponds to our protectedness.
+        assert!(self.permissions.all(|x| x.matches_protector(self.protected)));
+
```

Comment:
We can remove this again, because `transition` "obviously" changes neither the protector nor the "protected or not" variant of the permission.